modules/cudaarithm/src/cuda/transpose.cu

```diff
@@ -56,40 +56,136 @@ using namespace cv;
 using namespace cv::cuda;
 using namespace cv::cudev;
 
-void cv::cuda::transpose(InputArray _src, OutputArray _dst, Stream& stream)
-{
-    GpuMat src = getInputMat(_src, stream);
+#define USE_NPP_STREAM_CONTEXT (NPP_VERSION >= (10 * 1000 + 1 * 100 + 0))
```

Comment:
I think it would be better to define this globally somewhere although not sure where, maybe [modules/cudev/include/opencv2/cudev/common.hpp](https://github.com/cudawarped/opencv_contrib/blob/e40c43d96a22edaa3fbb880e957a3753938dc4f0/modules/cudev/include/opencv2/cudev/common.hpp#L61).